node/v2/handler.js

```diff
@@ -183,14 +183,21 @@ TChannelV2Handler.prototype.handleInitResponse = function handleInitResponse(res
 
 TChannelV2Handler.prototype.handleCallRequest = function handleCallRequest(reqFrame) {
     var self = this;
+
     if (self.remoteName === null) {
-        return self.errorEvent.emit(self, errors.CallReqBeforeInitReqError());
+        self.errorEvent.emit(self, errors.CallReqBeforeInitReqError());
+        return;
     }
 
     var req = self.buildInRequest(reqFrame);
 
-    if (self.incomingRequestInvalid(reqFrame, req)) {
-        return null;
+    var err = self.checkCallReqFrame(reqFrame);
+    if (err) {
+        req.res = self.buildOutResponse(req);
+        self.sendErrorFrame(
+            req.res, 'ProtocolError', err.message
```

Comment:
Will do, good call; it's something I handle better later by simply emitting the err on req.errorEvent, which gets hooked up to such classification.  However I didn't want to add all of that in this PR.

Confirmed classification of arg1 over length, as of 4b45dec errors.js:578 has it as a `BadRequest`.